index.js

```diff
@@ -166,30 +157,33 @@ const compile = function(schema, root, reporter, opts, scope, basePathRoot) {
       fun.write('errors++')
       if (reporter === true) {
         fun.write('if (validate.errors === null) validate.errors = []')
+        const errorObject = { field: prop || name, message: msg }
         if (verbose) {
+          const type = node.type || 'any'
+          Object.assign(errorObject, { type, schemaPath: toPointer(schemaPath) })
           fun.write(
-            'validate.errors.push({field:%s,message:%s,value:%s,type:%s,schemaPath:%s})',
-            formatName(prop || name),
-            JSON.stringify(msg),
-            value || name,
-            JSON.stringify(node.type || 'any'),
-            JSON.stringify(schemaPath)
+            'validate.errors.push({ ...%s, value: %s })',
+            JSON.stringify(errorObject),
+            value || name
           )
         } else {
-          fun.write(
-            'validate.errors.push({field:%s,message:%s})',
-            formatName(prop || name),
-            JSON.stringify(msg)
-          )
+          fun.write('validate.errors.push(%s)', JSON.stringify(errorObject))
         }
       }
       if (!allErrors) fun.write('return false')
     }
 
+    const fail = (msg, value) => {
+      const comment = value !== undefined ? ` ${JSON.stringify(value)}` : ''
+      throw new Error(`${msg}${comment} at #${toPointer(schemaPath)}`)
+    }
+    const enforce = (ok, ...args) => ok || fail(...args)
+    const validationRequired = (msg) => enforce(!requireValidation, `[requireValidation] ${msg}`)
```

Comment:
Unfriendly var name, easily confused with `requireValidation`. Suggestions: `failIfValidationRequired` `assertWithValidation` `validationFail`

Reply:
Renamed to `enforceValidation` to match with `enforce`.